### PR TITLE
Support @deprecated

### PR DESCRIPTION
**Pull request by newhouse**
Fixes https://github.com/anvilco/spectaql/issues/18

image

image

### CODE REVIEW

File: app/views/partials/json-schema/enum.hbs [hunk 1]
```diff
@@ -13,13 +13,18 @@
...
-              {{md description}}
-            {{/if}}
-          </td>
+          <td>{{md (codify enum.[0]) addClass=(ternary (and isDeprecated (not @root.info.x-hideIsDeprecated)) "swagger-operation-deprecated" false)}}</td>
+          {{! The Description column }}
+          <td>{{ md (
+              concat
+                (stripTrailing description ".")
+                (ternary (and deprecationReason (not @root.info.x-hideDeprecationReason))
+                  (concat "Deprecation Reason: " deprecationReason)
```
Comment (benogle) [inline app/views/partials/json-schema/enum.hbs:22]: Can you wrap the reason in a span with a class so styles can be overridden?

File: app/views/partials/json-schema/properties.hbs [hunk 2]
```diff
@@ -42,16 +43,21 @@
...
-            --}}
-          {{/if}}
+          {{ md (
+            concat
+              (ternary description
+                (stripTrailing description ".")
+                ""
+              )
+              (ternary (and deprecationReason (not @root.info.x-hideDeprecationReason))
+                (concat "Deprecation Reason: " deprecationReason)
```
Comment (benogle) [inline app/views/partials/json-schema/properties.hbs:55]: Here too, can you wrap the reason in a span with a class so styles can be overridden?

File: app/views/partials/swagger/operations/operation.hbs [hunk 3]
```diff
@@ -25,15 +25,24 @@
...
       <div class="doc-copy">
         <section class="swagger-operation-description operation-description">
-          {{md description}}
+          {{ md (
+              concat
+                (stripTrailing description ".")
+                (ternary (and deprecationReason (not @root.info.x-hideDeprecationReason))
+                  (concat "Deprecation Reason: " deprecationReason)
+                  ""
+                )
```
Comment (benogle) [inline app/views/partials/swagger/operations/operation.hbs:42]: Can you put the reason in this case above the description with its own section & class? I think just making it italic by default would be sufficient. e.g. like this:

screen_shot_2021-07-14_at_3 42 33_pm

**Comment by newhouse** [comment]
try now @benogle

**Review by benogle — APPROVED** [review]
👍

**Pull request merged**